Node.js/GettingStarted/nodejs_batch_client_sample.js

```diff
@@ -0,0 +1,188 @@
+// Initializing node.js modules
+const sleepInterval = 4000;
+const batch = require('azure-batch');
+const https = require('https');
+const fs = require('fs');
+const uuid = require('node-uuid');
+const sleep = require('system-sleep');
+
+// Setting up variables specific to Batch & storage account
+var accountName = '<azure-batch-account-name>';
+var accountKey = '<account-key-downloaded>';
+var accountUrl = '<account-url>'
+var sh_url = "Shell script SAS URI";
+var scriptURI = "Python script SAS URI";
+
+// Max retries
+const maxRetriesVal = 100;
+// Job ID 
+var jobId = "processcsvjob"; 
+
+// Create Batch credentials object using account name and account key
+var credentials = new batch.SharedKeyCredentials(accountName,accountKey);    
+
+// Create Batch service client
+var batchClient = new batch.ServiceClient(credentials,accountUrl);
+
+var now = new Date();
+var poolId = "processcsv_"+ now.getFullYear() + now.getMonth() + now.getDay() + now.getHours() + now.getSeconds()
+
+console.log("Creating pool with ID" + poolId);
+
+var imgRef = {publisher:"Canonical",
+                offer:"UbuntuServer",
+                sku:"14.04.2-LTS",
+                version:"latest"}
+var vmConfig = {imageReference:imgRef,
+    nodeAgentSKUId:"batch.node.ubuntu 14.04"}
+
+var numVms = 4;
+var vmSize = "STANDARD_A1";
+var clientPoolRequestID = uuid.v4();
+
+var poolConfig = {id:poolId, 
+    displayName:"Processing csv files",
+    vmSize:vmSize,
+    virtualMachineConfiguration:vmConfig,
+    targetDedicated:numVms,
+    enableAutoScale:false }
+var isPoolCreationError = false;
+var pool = batchClient.pool.add(poolConfig,function(error,result,request,response){
+    if(error !== null)
+    {
+        console.log(error.response);
+        isPoolCreationError = true;
+    }
+ });
+console.log("Submitted request to create pool...");
+
+var isPoolStateActive = false;
+var numRetries = 0;
+var jobCreated = false;
+
+while(isPoolStateActive == false && isPoolCreationError == false && numRetries < maxRetriesVal)
```

Comment:
@shwetams I totally sympathise with the lack of documentation and the importance of getting something working in order to understand it (and to satisfy the customer), and we really appreciate you contributing it back to the community.  We need to balance showing how to use the API functions with showing recommended structures for using them.  I appreciate your time is limited, and that there are no samples you can refer to (which is why I'm delighted that you've contributed this one), so there's nothing I can point to you to model the structure on, and you don't want to be saddled with too much polishing.  I'm happy to work with you on the callback/promise approach (I'm no Node.js expert myself, so it will be good learning for me too), or we could take the PR as it stands and do a polishing pass 'in house'.  What works best for you?  cc @darylmsft @matthchr @xingwu1